rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/README.asciidoc

```diff
@@ -115,6 +117,51 @@ Unless only legacy test runner `features` are required, a `reason` must also be
 so all the tests in a file will be skipped if either any requirement fails or any skip condition applies regardless
 if defined in `setup` and `teardown`.
 
+[[capabilities_check]]
+=== Require or skip API capabilities
+
+As opposed to <<cluster_features,cluster features>>, which are aimed at doing fast checks internally,
+the capabilities API allows external clients to ask an elasticsearch cluster what it supports in terms of
+particular endpoints, query parameters and other arbitrary capabilities.
+
+Only if every node in the cluster supports the requested path and method with all parameters and capabilities,
+the capabilities check passes successfully. Capabilities checks can be done both for `skip` and `requires`
+prerequisites. In either case, the _capabilities_ test runner feature must be required to allow
+other test runners to skip tests if they do not support the capabilities API yet.
+
+....
+    "Parent":
+     - requires:
+          capabilities:
+            - method: GET
+              path: /_api
+              parameters: [param1, param2]
+              capabilities: [cap1, cap2]
+          test_runner_feature: [capabilities]
+          reason: Capability required to run test
+     - do:
+       ... test definitions ...
+....
+
+Note: If planning to `skip` on capabilities, keep in mind this might lead to unexpected results in _mixed cluster_
+tests. A test is only skipped if all nodes support the requested capabilities, in _mixed clusters_ this might not be
+the case: you might randomly hit a node that actually supports what you intended to skip on.
```

Comment:
I think this might need a bit more explanation - 'you might randomly hit a node that actually supports what you intended to skip on', what is the effect of this? Why does this matter when the capabilities check is across the whole cluster?

Reply:
> and you might randomly hit one of the nodes

Worth mentioning this is "if there is no internal coordination on the cluster feature"? Or would this makes the text to complicated?

Reply:
I think we need some info like 'if not all nodes support the capabilities API itself, all capabilities are unsupported'

Reply:
> I think we need some info like 'if not all nodes support the capabilities API itself, all capabilities are unsupported'

Do you really think that needs repetition @thecoop ? That should be clear based on the following just a few sentences above.

> Only if every node in the cluster supports the requested path and method with all parameters and capabilities,
the capabilities check passes successfully

Reply:
@ldematte I wouldn't want to complicate this further talking about cluster features.

Reply:
I think there is a use case (tests for previous behaviour)...

Reply:
Yes, that's all right